mdata/aggmetric_test.go

```diff
@@ -302,143 +300,29 @@ func TestAggMetricDropFirstChunk(t *testing.T) {
-	chunkMaxStale := uint32(3600)
-	metricMaxStale := uint32(21600)
-	mockstore.Reset()
-	mockstore.Drop = true
-	defer func() {
-		mockstore.Drop = false
-	}()
-	SetSingleAgg(conf.Avg, conf.Min, conf.Max)
-	SetSingleSchema(
-		conf.NewRetentionMT(1, 84600, 600, 5, true),
-		conf.NewRetentionMT(300, 84600, 24*3600, 2, true),
-	)
-
-	cluster.Init("default", "test", time.Now(), "http", 6060)
-
-	keys := make([]schema.MKey, 1000)
-	for i := 0; i < 1000; i++ {
-		keys[i] = test.GetMKey(i)
-	}
-
-	metrics := NewAggMetrics(mockstore, &cache.MockCache{}, false, chunkMaxStale, metricMaxStale, 0)
-
-	maxT := uint32(1200)
-	for t := uint32(1); t < maxT; t += 10 {
-		for metricI := 0; metricI < 1000; metricI++ {
-			k := keys[metricI]
-			m := metrics.GetOrCreate(k, 0, 0)
-			m.Add(t, float64(t))
-		}
-	}
-}
-
-func BenchmarkAggMetrics10kSeries2Chunks10kQueueSize(b *testing.B) {
-	chunkMaxStale := uint32(3600)
-	metricMaxStale := uint32(21600)
+func BenchmarkAggMetricAdd(b *testing.B) {
 	mockstore.Reset()
 	mockstore.Drop = true
 	defer func() {
 		mockstore.Drop = false
 	}()
-	SetSingleAgg(conf.Avg, conf.Min, conf.Max)
-	SetSingleSchema(
-		conf.NewRetentionMT(1, 84600, 600, 5, true),
-		conf.NewRetentionMT(300, 84600, 24*3600, 2, true),
-	)
 
 	cluster.Init("default", "test", time.Now(), "http", 6060)
 
-	keys := make([]schema.MKey, 1000)
-	for i := 0; i < 1000; i++ {
-		keys[i] = test.GetMKey(i)
+	retentions := conf.Retentions{
+		{
+			SecondsPerPoint: 10,
+			NumberOfPoints:  10e9, // TTL
+			ChunkSpan:       1800, // 30 min. contains 180 points at 10s resolution
+			NumChunks:       1,
+			Ready:           true,
+		},
 	}
 
-	metrics := NewAggMetrics(mockstore, &cache.MockCache{}, false, chunkMaxStale, metricMaxStale, 0)
-
-	maxT := uint32(1200)
-	for t := uint32(1); t < maxT; t += 10 {
-		for metricI := 0; metricI < 10000; metricI++ {
-			k := keys[metricI]
-			m := metrics.GetOrCreate(k, 0, 0)
-			m.Add(t, float64(t))
-		}
-	}
-}
+	metric := NewAggMetric(mockstore, &cache.MockCache{}, test.GetAMKey(0), retentions, 0, nil, false)
 
-func BenchmarkAggMetrics100kSeries2Chunks100kQueueSize(b *testing.B) {
-	chunkMaxStale := uint32(3600)
-	metricMaxStale := uint32(21600)
-	mockstore.Reset()
-	mockstore.Drop = true
-	defer func() {
-		mockstore.Drop = false
-	}()
-	SetSingleAgg(conf.Avg, conf.Min, conf.Max)
-	SetSingleSchema(
-		conf.NewRetentionMT(1, 84600, 600, 5, true),
-		conf.NewRetentionMT(300, 84600, 24*3600, 2, true),
-	)
-
-	cluster.Init("default", "test", time.Now(), "http", 6060)
-
-	keys := make([]schema.MKey, 1000)
-	for i := 0; i < 1000; i++ {
-		keys[i] = test.GetMKey(i)
-	}
-
-	metrics := NewAggMetrics(mockstore, &cache.MockCache{}, false, chunkMaxStale, metricMaxStale, 0)
-
-	maxT := uint32(1200)
-	for t := uint32(1); t < maxT; t += 10 {
-		for metricI := 0; metricI < 100000; metricI++ {
-			k := keys[metricI]
-			m := metrics.GetOrCreate(k, 0, 0)
-			m.Add(t, float64(t))
-		}
+	max := uint32(b.N*10 + 1)
```

Comment:
would a `b.resetCounter()` be appropriate here? as we don't want to measure the time of the above steps

Reply:
not needed. above steps are negligible